src/main/java/school/hei/patrimoine/compiler/CasFileCompiler.java

```diff
@@ -0,0 +1,68 @@
+package school.hei.patrimoine.compiler;
+
+import static javax.tools.ToolProvider.getSystemJavaCompiler;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Path;
+import java.util.function.Function;
+import lombok.SneakyThrows;
+
+public class CasFileCompiler implements Function<String, Class<?>> {
+
+  private static final String COMPILE_DIR_NAME =
+      System.getProperty("user.home") + "/.patrimoine/compile";
```

Comment:
user.home could be put in a static final variable to avoid querying it from system att ( all the time)